diff -Naur busybox-1.24.2-orig/miscutils/man.c busybox-1.24.2/miscutils/man.c
--- busybox-1.24.2-orig/miscutils/man.c	2016-06-11 14:18:09.601383806 +0200
+++ busybox-1.24.2/miscutils/man.c	2016-06-11 16:29:41.505266651 +0200
@@ -4,7 +4,7 @@
  */
 
 //usage:#define man_trivial_usage
-//usage:       "[-aw] [MANPAGE]..."
+//usage:       "[-aw] [[SECTION] [MANPAGE]...]..."
 //usage:#define man_full_usage "\n\n"
 //usage:       "Format and display manual page\n"
 //usage:     "\n	-a	Display all pages"
@@ -190,6 +190,7 @@
 	char *sec_list;
 	char *cur_path, *cur_sect;
 	char **man_path_list;
+	char *sect;
 	int count_mp;
 	int cur_mp;
 	int opt, not_found;
@@ -254,10 +255,15 @@
 	}
 
 	not_found = 0;
+	sect = NULL;
 	do { /* for each argv[] */
 		int found = 0;
 		cur_mp = 0;
 
+		if (isdigit(**argv) && argv[1] != NULL) {
+			sect = *argv;
+			goto next_arg;
+		}
 		if (strchr(*argv, '/')) {
 			found = show_manpage(pager, *argv, /*man:*/ 1, 0);
 			goto check_found;
@@ -267,19 +273,27 @@
 			cur_sect = sec_list;
 			do { /* for each section */
 				char *next_sect = strchrnul(cur_sect, ':');
-				int sect_len = next_sect - cur_sect;
+				int file_sect_len = next_sect - cur_sect;
+				int dir_sect_len = strspn(cur_sect, "0123456789");
 				char *man_filename;
 				int cat0man1 = 0;
 
+				if (sect != NULL) {
+					if (strcmp(cur_sect, sect) != 0
+					 && strncmp(cur_sect, sect, dir_sect_len) != 0) {
+						goto next_sect;
+					}
+				}
+
 				/* Search for cat, then man page */
 				while (cat0man1 < 2) {
 					int found_here;
 					man_filename = xasprintf("%s/%s%.*s/%s.%.*s",
 							cur_path,
 							"cat\0man" + (cat0man1 * 4),
-							sect_len, cur_sect,
+							dir_sect_len, cur_sect,
 							*argv,
-							sect_len, cur_sect);
+							file_sect_len, cur_sect);
 					found_here = show_manpage(pager, man_filename, cat0man1, 0);
 					found |= found_here;
 					cat0man1 += found_here + 1;
@@ -288,6 +302,7 @@
 
 				if (found && !(opt & OPT_a))
 					goto next_arg;
+ next_sect:
 				cur_sect = next_sect;
 				while (*cur_sect == ':')
 					cur_sect++;
